build.gradle

```
@@ -35,27 +33,164 @@ shadowJar {
+    compileOnly "org.embulk:embulk-spi:$embulkVersion"
+
+    compile('org.embulk:embulk-util-config:0.3.1') {
+        // Conflict with Embulk Core
+        exclude group: 'com.fasterxml.jackson.core', module: 'jackson-annotations'
+        exclude group: 'com.fasterxml.jackson.core', module: 'jackson-core'
+        exclude group: 'com.fasterxml.jackson.core', module: 'jackson-databind'
+        exclude group: 'com.fasterxml.jackson.datatype', module: 'jackson-datatype-jdk8'
+        exclude group: 'javax.validation', module: 'validation-api'
+    }
+    compile 'com.fasterxml.jackson.core:jackson-core:2.6.7'
+    compile 'com.fasterxml.jackson.core:jackson-annotations:2.6.7'
+    compile 'com.fasterxml.jackson.core:jackson-databind:2.6.7'
+    compile 'com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.6.7'
+    compile 'javax.validation:validation-api:1.1.0.Final'
+
+    compile "org.embulk:embulk-util-file:0.1.3"
+    compile('org.embulk:embulk-util-retryhelper:0.8.1') {
+        exclude group: 'org.slf4j', module: 'slf4j-api'
+    }
 
     compile "com.google.cloud:google-cloud-storage:1.56.0"
 
     testCompile "junit:junit:4.12"
-    testCompile "org.embulk:embulk-core:0.9.12:tests"
-    testCompile "org.embulk:embulk-standards:0.9.12"
+    testCompile "org.embulk:embulk-core:$embulkVersion"
+    testCompile "org.embulk:embulk-core:$embulkVersion:tests"
+    testCompile "org.embulk:embulk-deps:$embulkVersion"
+    testCompile "org.embulk:embulk-junit4:$embulkVersion"
     testCompile "org.mockito:mockito-core:2.18.3"
+    testCompile 'com.google.code.findbugs:annotations:3.0.0'
+    testCompile "org.embulk:embulk-formatter-csv:0.10.36"
+    testCompile "org.embulk:embulk-output-file:0.10.36"
+    testCompile "org.embulk:embulk-parser-csv:0.10.36"
+}
+
+embulkPlugin {
+    mainClass = "org.embulk.input.gcs.GcsFileInputPlugin"
+    category = "input"
+    type = "gcs"
+}
+
+javadoc {
+    options {
+        locale = 'en_US'
+        encoding = 'UTF-8'
+    }
+}
+
+jar {
+    from rootProject.file("LICENSE.txt")
+}
+
+sourcesJar {
+    from rootProject.file("LICENSE.txt")
+}
+
+javadocJar {
+    from rootProject.file("LICENSE.txt")
+}
+
+publishing {
+    publications {
+        maven(MavenPublication) {
+            groupId = project.group
+            artifactId = project.name
+
+            from components.java
+
+            pom {  // https://central.sonatype.org/pages/requirements.html
+                packaging "jar"
+
+                name = project.name
+                description = project.description
+                url = "https://www.embulk.org/"
+
+                developers {
+                    developer {
+                        name = "Satoshi Akama"
+                        email = "satoshiakama@gmail.com"
+                    }
+                    developer {
+                        name = "John Luong"
+                        email = "jluong@treasure-data.com"
+                    }
```

From the Git log, let's go with this :

```suggestion
                    developer {
                        name = "Satoshi Akama"
                        email = "satoshiakama@gmail.com"
                    }
                    developer {
                        name = "Muga Nishizawa"
                        email = "muga.nishizawa@gmail.com"
                    }
                    developer {
                        name = "Naotoshi Seo"
                        email = "sonots@gmail.com"
                    }
                    developer {
                        name = "Robert Nguyen"
                        email = "ng.hung83@gmail.com"
                    }
                    developer {
                        name = "Hieu Duong"
                        email = "duongminhhieu89@gmail.com"
                    }
                    developer {
                        name = "John Luong"
                        email = "jluong@treasure-data.com"
                    }
```